src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,141 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            first = newNode;
+        } else {
+            newNode.prev = last;
+            last.next = newNode;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, first);
+            first.prev = newNode;
+            first = newNode;
+            size++;
+        } else {
+            addNode(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> result = getNodeByIndex(index);
+        return result.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> result = getNodeByIndex(index);
+        T oldValue = result.value;
+        result.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        return delNode(getNodeByIndex(index));
     }
 
     @Override
-    public boolean remove(T object) {
+    public boolean remove(T value) {
+        for (Node<T> obj = first; obj != null; obj = obj.next) {
+            if ((obj.value == value || (obj.value != null && obj.value.equals(value)))) {
+                delNode(obj);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private T delNode(Node<T> node) {
```

Comment:
```suggestion
    private T unlink(Node<T> node) {
```
as described in common mistakes